type.go

```diff
@@ -646,3 +841,115 @@ func (d *DataManager) GetFreeFloatings(param * FreeFloatingRequestParameter) (fr
 	}
 	return resp, nil
 }
+
+func (d *DataManager) InitStopPoint(stopPoints map[string]StopPoint) {
+	d.vehicleOccupanciesMutex.Lock()
+	defer d.vehicleOccupanciesMutex.Unlock()
+
+	d.stopPoints = &stopPoints
+	fmt.Println("*** stopPoints size: ", len(*d.stopPoints))
+	d.lastVehicleOccupanciesUpdate = time.Now()
+}
+
+func (d *DataManager) GetStopId(name string, sens int) (id string) {
+	d.vehicleOccupanciesMutex.RLock()
+	defer d.vehicleOccupanciesMutex.RUnlock()
+
+	key := name + strconv.Itoa(sens)
+	return (*d.stopPoints)[key].Id
+}
+
+func (d *DataManager) InitCourse(courses map[string][]Course) {
+	d.vehicleOccupanciesMutex.Lock()
+	defer d.vehicleOccupanciesMutex.Unlock()
+
+	d.courses = &courses
+	fmt.Println("*** courses size: ", len(*d.courses))
+	d.lastVehicleOccupanciesUpdate = time.Now()
+}
+
+func (d *DataManager) InitRouteSchedule(routeSchedules []RouteSchedule) {
+	d.vehicleOccupanciesMutex.Lock()
+	defer d.vehicleOccupanciesMutex.Unlock()
+
+	d.routeSchedules = &routeSchedules
+	fmt.Println("*** routeSchedules size: ", len(*d.routeSchedules))
+	d.lastVehicleOccupanciesUpdate = time.Now()
+}
+
+func (d *DataManager) UpdateVehicleOccupancies(vehicleOccupancies map[int]VehicleOccupancy) {
+	d.vehicleOccupanciesMutex.Lock()
+	defer d.vehicleOccupanciesMutex.Unlock()
+
+	d.vehicleOccupancies = &vehicleOccupancies
+	fmt.Println("*** vehicleOccupancies size: ", len(*d.vehicleOccupancies))
+	d.lastVehicleOccupanciesUpdate = time.Now()
+}
+
+func (d *DataManager) GetLastVehicleOccupanciesDataUpdate() time.Time {
+	d.vehicleOccupanciesMutex.RLock()
+	defer d.vehicleOccupanciesMutex.RUnlock()
+
+	return d.lastVehicleOccupanciesUpdate
+}
+
+func (d *DataManager) GetCourseFirstTime(prediction Prediction) (date_time time.Time, e error) {
+	d.vehicleOccupanciesMutex.RLock()
+	defer d.vehicleOccupanciesMutex.RUnlock()
+	for _, course := range (*d.courses)[prediction.LineCode] {
+		if prediction.Course == course.Course && int(prediction.Date.Weekday()) == course.Dow {
+			return course.FirstTime, nil
+		}
+	}
+	return time.Now(), fmt.Errorf("No corresponding data found")
+}
+
+func (d *DataManager) GetVehicleJourneyId(predict Prediction, dataTime time.Time) (vj string) {
+	d.vehicleOccupanciesMutex.RLock()
+	defer d.vehicleOccupanciesMutex.RUnlock()
+	for _, rs := range (*d.routeSchedules) {
```

Comment:
If I understand correctly : this looks for a `rs` whose `rs.DateTime` is within two minutes of `dataTime`, and returns "" if no such vehicle_journey is found.

I think the "two minutes" is a bit arbitrary and may be too small or too big depending on the data.

I suggest replacing this by returning the `rs` such that `abs(rs.DateTime - dataTime)` is minimal.
Something like : 
```suggestion
min_distance = +inf
result = nil   
for _, rs := range (*d.routeSchedules) {
  if rs.Departure == true &&
  predict.LineCode == rs.LineCode &&
  predict.Sens == rs.Sens &&
  abs(rs.DateTime - dataTime) < min_distance
  {
      min_distance = abs(rs.DateTime - dataTime) 
      result = rs.VehicleJourneyId
  }
}
return result
```